### ISSUE TITLE
[InputAccessoryView] Can't specify keyboard-conditional bottom padding

### ISSUE THREAD

**Issue by Ashoat** [facebook/react-native#20157]
## Environment
```
React Native Environment Info:
  System:
    OS: macOS High Sierra 10.13.5
    CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 25.07 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 9.3.0 - ~/.nvm/versions/node/v9.3.0/bin/node
    Yarn: 1.6.0 - /usr/local/bin/yarn
    npm: 6.1.0 - ~/.nvm/versions/node/v9.3.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 11.4, macOS 10.13, tvOS 11.4, watchOS 4.3
    Android SDK:
      Build Tools: 23.0.1, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.3
      API Levels: 23, 25, 26
  IDEs:
    Android Studio: 3.1 AI-173.4819257
    Xcode: 9.4.1/9F2000 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.4.1 => 16.4.1
    react-native: ^0.56.0 => 0.56.0
```

## Description
If you have a bottom tab bar, you want the `InputAccessoryView` to appear above it when the keyboard is collapsed. However, when the keyboard is expanded, you want the `InputAccessoryView` to appear directly above the keyboard.

This sort of "conditional" bottom padding is currently impossible with `InputAccessoryView`. It seems that when the keyboard is expanded, the distance between the bottom of the screen and the `InputAccessoryView` is fixed. Attempting to change this distance with `height`, `padding`, or `margin` style properties does not seem possible.

## Reproducible Demo
I've created a minimal repro [here](https://github.com/Ashoat/TabsWithInputAccessoryView).

**Comment by jakebloom** [comment]
@cpojer What is the status on this? Does the linked issue being closed indicate that this has been resolved?

**Comment by cpojer** [comment]
No, it seems like this issue still exists. Please feel free to send a Pull Request with a fix.

**Comment by digitalbreed** [comment]
Until there is a fix, is there any workaround?

**Comment by darrylyoung** [comment]
I'm also looking for a solution to this problem. I want an input and button sat at the bottom of my screen (on top of the tab bar) and when the input is in focus, the keyboard slides up and those elements slide up with it. The _focused_ position works as expected but when the keyboard is collapsed, the input and button completely cover the tab bar.

Are there any workarounds for this while we're hoping for a proper solution? Did you have any luck, @digitalbreed / @Ashoat?

**Comment by Ashoat** [comment]
No, no solutions exist to my knowledge. You can avoid using `KeyboardAccessoryView` altogether, but that means you can’t pan the keyboard up/down... just press to open/close.

I’ve seen [this non-React-Native solution](https://github.com/iAmrMohamed/AMKeyboardFrameTracker), but it seems hard to port to React Native, and feels a little hacky.

**Comment by jamesxabregas** [comment]
I can confirm that this is still technically a current bug in React Native 0.63.3 and iOS 14.1. However after much consideration of this issue I have come to the conclusion that fixing this issue would like not provide a desierable UX design in most cases anyway. My app also uses a bottom tab bar, but after mocking out what my UI would like like with a bottom tab bar, open keryboard, and an InputAccessory I realised it would actually look very weird. In fact I looked through a number of popular apps that leverage an InputAccessoryView, (Facebook's create post screen is a common one) and in every case the screen that uses the InputAccessoryView is always a modal view that displays over the entire screen and hides the bottom tab bar. I assume that this is likely because displaying a bottom tab bar and an input accessory provides for a very confusing and clunky user expereince.

**Comment by gitVera** [comment]
https://reactnative.dev/docs/keyboard 
you can add listeners and hide tabbar if keyboard is active

**Comment by lucianobracco-geojam** [comment]
Still happening

**Comment by MarcusDelvecchio** [comment]
Still having issues with this

**Comment by kirillzyusko** [comment]
@Ashoat Maybe it's doable with https://github.com/kirillzyusko/react-native-keyboard-controller and [KeyboardStickyView](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/components/keyboard-sticky-view) component?

You can specify `offset` property for `closed`/`opened` state and it should work as expected 👀 It also support interactive keyboard dismissal so I think worth to try it out!

**Comment by Ashoat** [comment]
Looks super cool!! Will take a look at it when I get a chance. Thanks for sharing @kirillzyusko!

**Comment by mozzius** [comment]
This has the downside of not supporting interactive dismissal starting at the component, see the issue here: https://github.com/kirillzyusko/react-native-keyboard-controller/issues/250

As it stands, you need to choose between having interactive dismiss or being able to set the position of the inputaccessoryview

Edit: react-native-keyboard-controller now supports this!

**Comment by kirillzyusko** [comment]
> This has the downside of not supporting interactive dismissal starting at the component, see the issue here: https://github.com/kirillzyusko/react-native-keyboard-controller/issues/250
>
> As it stands, you need to choose between having interactive dismiss or being able to set the position of the inputaccessoryview

Yeah, I'm planning to fix this problem in the near future 👀

**Comment by pmcb99** [comment]
crazy how this hasn't been dealt with natively without a community package. React Native ecosystem is cooked
